core/tests/protractor/subscriptionsFlow.js

```diff
@@ -0,0 +1,124 @@
+ * and mobile.
+ */
+
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var waitFor = require('../protractor_utils/waitFor.js');
+
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var CreatorDashboardPage =
+  require('../protractor_utils/CreatorDashboardPage.js');
+var PreferencesPage = require('../protractor_utils/PreferencesPage.js');
+var SubscriptionDashboardPage =
+  require('../protractor_utils/SubscriptionDashboardPage.js');
+
+describe('Subscriptions functionality on mobile', function() {
+  var creatorDashboardPage = null;
+  var preferencesPage = null;
+  var subscriptionDashboardPage = null;
+
+  beforeEach(function() {
+    creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+    preferencesPage = new PreferencesPage.PreferencesPage();
+    subscriptionDashboardPage = (
+      new SubscriptionDashboardPage.SubscriptionDashboardPage());
+  });
+
+  it('handle subscriptions to creators correctly', function() {
+    // Create two creators.
+    users.createUser('creator1Id@subscriptions.com', 'creator1Idsubscriptions');
+    users.createUser('creator2Id@subscriptions.com', 'creator2Idsubscriptions');
+
+    // Create a learner who subscribes to both the creators.
+    users.createUser('learner1@subscriptions.com', 'learner1subscriptions');
+    users.login('learner1@subscriptions.com');
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(
+      'creator1Idsubscriptions');
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(
+      'creator2Idsubscriptions');
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    preferencesPage.get();
+    preferencesPage.expectDisplayedFirstSubscriptionToBe('creator...');
+    preferencesPage.expectDisplayedLastSubscriptionToBe('creator...');
+    users.logout();
+
+    // Create a learner who subscribes to creator1Id and unsubscribes from the
+    // creator2Id.
+    users.createUser('learner2@subscriptions.com', 'learner2subscriptions');
+    users.login('learner2@subscriptions.com');
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(
+      'creator1Idsubscriptions');
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(
+      'creator2Idsubscriptions');
+
+    // Subscribe and then unsubscribe from the same user.
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    preferencesPage.get();
+    preferencesPage.expectSubscriptionCountToEqual(1);
+    preferencesPage.expectDisplayedFirstSubscriptionToBe('creator...');
+    users.logout();
+
+    // Verify there are 2 subscribers.
+    users.login('creator1Id@subscriptions.com');
+    // Go to the creator_dashboard.
+    creatorDashboardPage.get();
+    // There is no 'Create' button for small
```

Comment:
Nit: might want to use the full 80-char limit. This comment seems to be line-truncated pretty early for some reason.

Reply:
Done, thanks!